core-test/src/test/scala/org/bitcoins/core/crypto/SchnorrTest.scala

```diff
@@ -0,0 +1,166 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoin.NativeSecp256k1
+import org.bitcoins.core.util.{CryptoUtil, NumberUtil}
+import org.bitcoins.testkit.core.gen.{CryptoGenerators, NumberGenerator}
+import org.bitcoins.testkit.util.BitcoinSUnitTest
+import scodec.bits.ByteVector
+
+class SchnorrTest extends BitcoinSUnitTest {
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    generatorDrivenConfigNewCode
+
+  behavior of "NativeSecp256k1"
+
+  it must "act like a normal schnorrSign if the normal nonce is specified" in {
+    forAll(CryptoGenerators.nonZeroPrivKey, NumberGenerator.bytevector(32)) {
+      case (privKey, message) =>
+        val nonce = SchnorrNonce.fromBipSchnorr(privKey, message)
+
+        assert(nonce.bytes.toArray.exists(_ != 0.toByte))
+
+        val sigWithNonce =
+          NativeSecp256k1.schnorrSignWithNonce(message.toArray,
+                                               privKey.bytes.toArray,
+                                               nonce.bytes.toArray)
+
+        assert(
+          SchnorrDigitalSignature(ByteVector(sigWithNonce)) ==
+            Schnorr.signWithNonce(message, privKey, nonce)
+        )
+
+        val sig =
+          NativeSecp256k1.schnorrSign(message.toArray, privKey.bytes.toArray)
+
+        assert(
+          SchnorrDigitalSignature(ByteVector(sig)) ==
+            Schnorr.sign(message, privKey)
+        )
+
+        assert(ByteVector(sigWithNonce) == ByteVector(sig))
+    }
+  }
+
+  it must "correctly compute a Schnorr public key given an R value" in {
+    forAll(CryptoGenerators.nonZeroPrivKey, NumberGenerator.bytevector(32)) {
+      case (privKey, message) =>
+        val publicKey: ECPublicKey = privKey.publicKey
+        val messageArr: Array[Byte] = message.toArray
+
+        val nonce = SchnorrNonce.fromBipSchnorr(privKey, message)
+
+        assert(nonce.bytes.toArray.exists(_ != 0.toByte))
+
+        val rBytes = nonce.publicKey.bytes
```

Comment:
//add a self documenting comment of why the nonce's public key is the r value? 

I honestly don't understand why this is the case.

Reply:
In general, you are going to have to help 'narrate' me through this test case, I don't fully understand what's going on. Take me step by step through each assertion, and why it should be true, similar to what I kind of did for myself in the first test case in this file.

Reply:
I added these comments, lmk what you think, but RE this specific comment, that's just its definition (`R := nonce.publicKey`)